pom.xml

```diff
@@ -124,6 +129,45 @@
           </execution>
         </executions>
       </plugin>
+      <!-- Spotless plugin: keeps the code formatted following the google-java-styleguide -->
+      <plugin>
+        <groupId>com.diffplug.spotless</groupId>
+        <artifactId>spotless-maven-plugin</artifactId>
+        <version>2.40.0</version>
+        <executions>
+          <execution>
+            <goals>
+              <goal>check</goal>
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <formats>
+            <format>
+              <includes>
+                <include>*.md</include>
+                <include>.gitignore</include>
+              </includes>
+              <trimTrailingWhitespace/>
+              <endWithNewline/>
+              <indent>
+                <spaces>true</spaces>
+                <spacesPerTab>4</spacesPerTab>
+              </indent>
+            </format>
+          </formats>
```

Comment:
Is this `<formats>` section needed? If I understand it correctly that would only apply to `*.md` and `.gitignore` the way it is currently configured; and I am not sure if we should impose any formatting rules on them.

Reply:
Formatting rules are very weak on`. md` and `.gitignore`. They replace tab characters with 4 spaces and trim trailing whitespace. 

I think they are good general rules, but if you prefer to remove them, let me know.

Reply:
Ok, I just asked because this section looked like the section from the Spotless examples and it isn't needed for Java formatting.

The combination of `*.md` with `trimTrailingWhitespace` should be removed though because they can break Markdown formatting. Arguably the Spotless examples are also flawed (https://github.com/diffplug/spotless/issues/1877).